proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,1067 @@
+* A `processing directive` provides additional information to the client on how
+  the URI is to be interpreted. Two supported directives are:
+  * `alt=<xdstp:// URI>` expressing a fallback URI.
+  * `entry=<resource name>` providing an anchor referring to an inlined resource
+    for [list collections](#list). Resource names must be of the form `[a-zA-Z0-9_-\./]+`.
+
+Multiple directives are comma separated in the fragment component of the URI.
+The same directive may not appear more than once in a given URI.
+
+An example application of the above URI scheme is:
+
+`xdstp://some.control.plane/envoy.config.route.v3.RouteConfiguration/foo/bar?shard_id=1234&direction=inbound`
+
+We distinguish between two forms of `xdstp://` URIs:
+
+* Uniform Resource Locators (URLs) instructing a client on how to locate a given
+  xDS resource. URLs may include processing directives and [globs](#glob).
+
+* Uniform Resource Names (URNs) referring to a specific xDS resource. URNs do
+  not include processing directives or [globs](#glob). `xdstp://` URNs provide
+  self-contained cache keys.
+
+#### Proto3 expression
+
+`xdstp://` URIs are canonically encoded in `proto3` messages. The structure
+for this encoding is provided at https://github.com/cncf/xds/tree/main/xds/core/v3:
+
+* [URL](https://github.com/cncf/xds/blob/main/xds/core/v3/resource_locator.proto) definition
+* [URN](https://github.com/cncf/xds/blob/main/xds/core/v3/resource_name.proto) definition
+
+#### Normalization
+
+Two `xdstp://` URNs are considered equivalent if they match component-wise
+modulo context parameter ordering.
+
+#### Context parameters
+
+Context parameters in URNs presented by the client to the server will be
+composed from the following sources. Using an example of a URL
+`xdstp://some-authority/some.type/foo?xds.bar=baz`:
+
+* Static `Node`-derived context parameters. These are prefixed with
+  `xds.node.`. The set of `Node`-derived context parameters is specified in the
+  bootstrap on a per-resource type basis. The key reflects the `Node` proto3
+  field structure, e.g. `xds.node.locality.sub_zone=some_sub_zone`,
+  `xds.node.metadata.bar="a"`, `xds.node.user_agent_version=1.2.3`.
+  * Generally, values in the `Node` are converted from their proto3 value
+    to JSON following the [canonical
+    transformation](https://developers.google.com/protocol-buffers/docs/proto3#json).
+  * Both `xds.node.metadata.X` and
+    `xds.node.user_agent_build_version.metadata.X` permit directly referencing
+    a top-level metadata field `X` in a context parameter key. This does not
+    apply to nested fields in the metadata, e.g. `xds.node.metadata.bar`.
+  * `xds.node.user_agent_build_version.version` yields a string value composed
+    of `major.minor.patch` values, e.g. `"1.2.3"`.
+
+* Context parameters from the URL, in the above example
+  `xds.bar=baz`. These must not overlap with any other context parameter name,
```

Comment:
@markdroth should we put this in some namespace, e.g. `xds.params.` to future proof other parts of the `xds.` hierarchy in case we want them?